src/otx/algo/classification/torchvision_model.py

```diff
@@ -137,6 +139,12 @@ def __init__(
         self.softmax = nn.Softmax(dim=-1)
         self.loss = loss
 
+        self.explainer = ReciproCAMHook(
```

Comment:
What will happen if the user will run some unsupported model, e.g. one of `swin` models?

Reply:
What is the exact definition of the user you said? If the user is a kind of model developer who implements his own model, all we expect from him are implementing `OTXModel.forward_explain()` in the derived class.

> - In addition, it also does not fit the spirit of having this interface, `OTXModel.forward_explain()`. All we needed to obtain XAI predictions are implementing just this function `OTXModel.forward_explain()` in the derived class such as
https://github.com/openvinotoolkit/training_extensions/blob/99b1de22102bda57e76635631a1ee1751e414dd8/src/otx/algo/classification/torchvision_model.py#L310-L322

Reply:
I mean will some exception will raise when someone would run `swin` model with `explain_mode`? `ReciproCAM` does not support some models, such as `swin`.

Reply:
I think, Evgeny's point is that we have several algos in `src/otx/algo/hooks/recording_forward_hook.py` for different types of models. For transformer-based models like `swin` we should run `ViTReciproCAMHook`.

I think, I should revisit XAI in `torchvision_models` either way, so I will create if-else condition to chose proper algo for the proper architecture

Reply:
`ViTReciproCAMHook` most probably would not work with `swin` out of the box. `ViTReciproCAMHook` is applicable for vanila `ViT` architecture

Reply:
Oh, thank you for this point